src/xunit.netcore.extensions/Discoverers/ConditionalFactDiscoverer.cs

```diff
@@ -23,45 +24,71 @@ public ConditionalFactDiscoverer(IMessageSink diagnosticMessageSink) : base(diag
         public override IEnumerable<IXunitTestCase> Discover(
             ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
         {
-            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            string[] conditionMemberNames = factAttribute.GetConstructorArguments().FirstOrDefault() as string[];
+            IEnumerable<IXunitTestCase> testCases = base.Discover(discoveryOptions, testMethod, factAttribute);
+            return DiscoverConditionalTestCases(discoveryOptions, _diagnosticMessageSink, testMethod, testCases, conditionMemberNames);
+        }
 
-            string conditionMemberName = factAttribute.GetConstructorArguments().FirstOrDefault() as string;
-            Type declaringType = testMethodInfo.DeclaringType;
-            string[] symbols = conditionMemberName.Split('.');
+        // This helper method evaluates the given condition member names for a given set of test cases.
+        // If any condition member evaluates to 'false', the test cases are marked to be skipped.
+        // The skip reason is the collection of all the condition members that evalated to 'false'.
+        internal static IEnumerable<IXunitTestCase> DiscoverConditionalTestCases(
```

Comment:
If I were doing this fresh I'd put it in a distinct type, ConditionalTestDiscoverer, so it's able to be looked at independent of the Fact or Theory context (it felt a little weird when Theory called Fact).

I'm not sure that I'd do it as an enhancement; so this is fine.  Unless my comments here pushed you over the edge on an internal debate... in which case I'll smile supportively :smile:.